### DISCUSSION TITLE
Cloud variable length limit

### DISCUSSION THREAD

**Discussion by nico-tome** [TurboWarp/packager#381] [Q&A]
What is the precise length of a cloud variable and does it change if saved locally ?

**Comment by GarboMuffin** [comment]
Local cloud variables are only restricted by the amount of storage that your browser allows websites to use and local cloud variables do not need to be numbers.

**Reply by GarboMuffin** [reply]
Cloud variables that sync with other users currently have a 100000 character limit, must be numbers

**Reply by nico-tome** [reply]
OK, thanks. But I had three other questions about the cloud variables connected with a server, can we know when they will be reset precisely and if we change the host by changing the link must we do something in particular? And the last question is: if we put the same id for two different projects, if both projects have the same cloud variables will they be synchronized? (For example a "user" project and an "admin" project for a game)

**Reply by GarboMuffin** [reply]
> can we know when they will be reset precisely

This is intentionally not well defined. They are reset extremely often. Storing scores or saves here is practically impossible.

> if we change the host by changing the link must we do something in particular

The link has to point to a valid cloud variable server, that's all

> if we put the same id for two different projects, if both projects have the same cloud variables will they be synchronized

yes

**Reply by CST1229** [reply]
> This is intentionally not well defined. They are reset extremely often. Storing scores or saves here is practically impossible.

By the way, AFAIK they reset when either no one is cloud-connected to the project or when the server restarts.

**Reply by TheShovel** [reply]
You can use Sheeptester's primitive cloud server and put it on Herok to have no limits besides your storage and bandwith by the way.

**Reply by VBPROGER** [reply]
Here is link to cloud server, by the way: https://github.com/SheepTester/primitive-cloud-server

**Comment by MA-master-sec** [comment]
1e7-1 or 1e6-1
